xarray/core/dataset.py

```diff
@@ -6422,8 +6429,13 @@ def to_dask_dataframe(
             if not is_duck_dask_array(var._data):
                 var = var.chunk()
 
-            dask_array = var.set_dims(ordered_dims).chunk(self.chunks).data
-            series = dd.from_array(dask_array.reshape(-1), columns=[name])
+            if has_many_dims:
```

Comment:
Is this really that impactful, can we optimize `set_dims` instead?

Reply:
I'll think I'll save the has_many_dims paths for a future PR. I think it might introduce bugs if we don't consistently chunk with the same shape.